static/src/javascripts/projects/commercial/modules/messenger/type.ts

```diff
@@ -0,0 +1,20 @@
+import { isString } from '@guardian/libs';
+import fastdom from '../../../../lib/fastdom-promise';
+import type { RegisterListener } from '../messenger';
+
+const setType = (type: string, adSlot: Element) =>
+	fastdom.mutate(() => {
+		adSlot.classList.add(`ad-slot--${type}`);
+	});
+
+const init = (register: RegisterListener): void => {
+	register('type', (specs, ret, iframe) => {
+		const adSlot = iframe?.closest('.js-ad-slot');
+
+		if (adSlot && isString(specs)) {
+			void setType(specs, adSlot);
```

Comment:
`specs` -> `type` ? Difficult task - could we align the naming? :)

Reply:
Interesting one because `specs` comes up [a lot](https://github.com/guardian/frontend/search?q=specs) in the messenger code and it consistently represents the data transferred by an iframe to the page (the `value` field in our standard message format).

I'm happy with how consistently this is applied and I like the pattern of calling it `specs` until you've asserted it's of a certain type, but maybe `specs` could be more descriptive. In any case, I would prefer updating it across the board in a separate PR.

Reply:
Sure whatever you think is more consistent 👍 

Am I right in seeing that use of `specs` is overloaded though - i.e. it has varying 'shape' across the listeners?

Reply:
Yes exactly, so `specs: unknown` and the responsibility currently falls on the listener to narrow that into a type it understands.